ipet/parsing/StatisticReader.py

```diff
@@ -136,6 +136,33 @@ def turnIntoFloat(self, astring):
 ######################################################################################
 # DERIVED Classes
 
+class NodeNameReader(StatisticReader):
+    context = [Key.CONTEXT_LOGFILE, Key.CONTEXT_ERRFILE]
+
+    nodenameexp = re.compile('^Linux (\S*) .* GNU/Linux')
+    name = 'NodeNameReader'
+    datakey = Key.NodeName
+    nodename = None
+
+    def extractStatistic(self, line):
+        """ Read nodename data from slurminfo (first line)
+
+        Parameters
+        ----------
+        line, output of `uname -a` will be saved
+        """
+        if self.nodenameexp.search(line):
```

Comment:
You should probably use `match` and a match object from which you query the `groups()[0]` to make things faster.